enhancements/kube-apiserver/stability-point-to-point-network-check.md

```diff
@@ -0,0 +1,327 @@
+---
+title: point-to-point-network-check
+authors:
+  - "@deads2k"
+  - "@sanchezl"
+reviewers:
+approvers:
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+replaces:
+superseded-by:
+---
+
+# Stability: Point to Point Network Check
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance, 
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this? 
+
+## Summary
+
+There are a few key network connections that are important to the health of the cluster.
+We can create a CRD to store the the last success time for each point-to-point check.
+By watching for explicit failures and lack of updates, we can produce a picture of when a point-to-point check failed.
+This information could be correlated to e2e failures, alerts, events, logs, or other information feeds.
+
+## Motivation
+
+This attempts to identify failures of particular connections in a black-box manner that is independent of SDN, cloud,
+or cluster ingress.
+
+### Goals
+
+1. be independent of SDN, cloud, or cluster ingress
+2. Identify connection failure between each kas and each oas endpoint
+3. Identify connection failure between each kas and each etcd
+4. Identify connection failure between each kas and oas service IP
+5. Identify connection failure between each oas and each kas host IP
+6. Identify connection failure between each oas and each kas endpoint IP
+7. Identify connection failure between each oas and each etcd
+
+### Non-Goals
+
+1. DNS resolution, that would be distinct.
+
+## Proposal
```

Comment:
finish out this section.   Describe the process of which actor creates these resources and which actor will mark a particular one "dead" with an indication of the balance between latency and write load.